core/src/main/scala/com/databricks/labs/remorph/ApplicationContext.scala

```diff
@@ -70,4 +72,23 @@ trait ApplicationContext {
 
   def fileSetGenerator(dialect: String): FileSetGenerator =
     new FileSetGenerator(planParser(dialect), sqlGenerator, pySparkGenerator)
+
+  def generateDotFile(tableGraph: TableGraph, dstFile: File): Unit = {
+    val dotContent = new StringBuilder
+    dotContent.append("digraph lineage {\n")
+
+    tableGraph.getNodes.foreach { node =>
+      dotContent.append(s"""  "${node.tableDefinition.table}" [label="${node.tableDefinition.table}"];\n""")
+    }
+
+    tableGraph.getEdges.foreach { edge =>
+      edge.to.foreach { toNode =>
+        dotContent.append(s"""  "${edge.from.tableDefinition.table}" -> "${toNode.tableDefinition.table}";\n""")
+      }
+    }
+
+    dotContent.append("}\n")
+
+    Files.write(Paths.get(dstFile.getPath), dotContent.toString().getBytes)
+  }
```

Comment:
```suggestion
```

only declarative and dependency-binding code is allowed in `ApplicationContext`.